lib/cli/Cargo.toml

```diff
@@ -162,7 +162,7 @@ llvm = [
 debug = ["fern", "wasmer-wasi/logging"]
 disable-all-logging = ["wasmer-wasi/disable-all-logging"]
 headless = []
-headless-minimal = ["headless", "disable-all-logging", "wasi"]
+headless-minimal = ["headless", "disable-all-logging", "wasi", "compiler", "webc_runner", "static-artifact-create"]
```

Comment:
Is it really something we want, to have "headless-minimal" including "compiler". That doesn't sound minimal to me.

Reply:
It doesn't actually include any compiler, I'd personally like the `compiler` feature to be gone completely. If you look for `#[feature = "compiler"]` in the codebase, you can see that it's just controlling imports / exports.

Reply:
The `compiler` feature should be removed and always be enabled. There should only be feature flags for unnecessary dependencies and different targets. But otherwise, this can be replaced by `#[cfg(not(any(llvm, singlepass, cranelift)]` instead

Reply:
I don't think the "compiler" feature can be built for `wasm32-wasi` target

Reply:
It can, we currently have `--feature singlepass,cranelift` enabled on wasm32-wasi and both auto-enable the `compiler` feature

Reply:
I think this meshes with the complaints I had about compilers not being available, @ptitSeb .

I'd also like the basic compiler APIs to always be available, even if the environment doesn't actually have a valid implementation. 

(I don't know the APIs well yet, so not aware of complications this would cause)

Reply:
We want to include compilers in `wasm32-wasi`, because that way we can do the cloud compilation in the future.

But that doesn't mean that we should require "compiler" for the headless minimal

Reply:
```suggestion
headless-minimal = ["headless", "disable-all-logging", "wasi", "webc_runner", "static-artifact-create"]
```